DYDX-Bots/Roles_Monitor/src/agent.spec.ts

```diff
@@ -0,0 +1,346 @@
+import { FindingType, FindingSeverity, Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { utils } from "ethers";
+import { provideHandleTransaction } from "./agent";
+import NetworkManager from "./network";
+import { MODULE_IFACE } from "./utils";
+
+const testSender: string = createAddress("0xad");
+
+export const testRolesMap: Record<string, string> = {
+  "0x09659e0aa12c56a3dece97692f0f23fc8f2825f9ffaa3b75428e4726cdf2a814": "OWNER_ROLE", // "0xab11"
+  "0x5e620be0f7c67873fd132b7805b9bacd22770701962c4441433f6d572e09ac4c": "EPOCH_PARAMETERS_ROLE", // "0xab12"
+  "0xa4efcdbe0a4517e5f06e41c280c2685b29d3072f17b8ccb5ff92066a2a07f6de": "REWARDS_RATE_ROLE", // "0xab13"
+  "0x5a88a442e34c11687eb5944a8cd22b8238668156fa17d16580838a3b9e6c0c19": "BORROWER_ADMIN_ROLE", // "0xac21"
+  "0x71605e6bd15336ca6d608097000be252db45c93857c39479aaa8e66a40316928": "CLAIM_OPERATOR_ROLE", // "0xac22"
+  "0x15887ddf60e2e6b78e963b6dd4e93041dc8206450f1baa784eb0f6d704a44149": "STAKE_OPERATOR_ROLE", // "0xac23"
+  "0x514259bf2ff5d81df8bc4e02c421a05a287059ba14139e5620375140809e91e1": "DEBT_OPERATOR_ROLE", // "0xad31"
+  "0x54a8c0ab653c15bfb48b47fd011ba2b9617af01cb45cab344acd57c924d56798": "NONE", // "0x0000"
+  "0x673fbecda3bfdf1adbf9e5f658c0f5ee475166da8b4bcc9fd0b5d6dd0e82e0bd": "SLASHER_ROLE", // "0xad32"
+};
+
+const roleChangedTestCases: [string, string, string][] = [
+  [utils.keccak256("0xab11"), utils.keccak256("0xab12"), utils.keccak256("0xab13")],
+  [utils.keccak256("0xac21"), utils.keccak256("0xac22"), utils.keccak256("0xac23")],
+  [utils.keccak256("0xad31"), utils.keccak256("0x0000"), utils.keccak256("0xad32")],
+];
+// Format: [role, account, sender][]
+const grantedAndRevokedTestCases: [string, string, string][] = [
+  [utils.keccak256("0xab11"), createAddress("0xab2"), createAddress("0xab3")],
+  [utils.keccak256("0xab12"), createAddress("0xac2"), createAddress("0xac3")],
+  [utils.keccak256("0xab13"), createAddress("0xad2"), createAddress("0xad3")],
+  [utils.keccak256("0xac21"), createAddress("0xae2"), createAddress("0xae3")],
+  [utils.keccak256("0xac22"), createAddress("0xaf2"), createAddress("0xaf3")],
+  [utils.keccak256("0xac23"), createAddress("0xba2"), createAddress("0xba3")],
+];
```

Comment:
Just a comment for better coherence. You don't have to change this, the current version is fine and functional.

The Roles map structure is `< keccak256("someString"), someString > `,  It would be nice if `testRolesMap` followed that structure and was defined like :
```
{
keccak256("ROLE_ONE"): "ROLE_ONE",
keccak256("ROLE_TWO)":  "ROLE_TWO",
 ...
}
```
Then used the same roles strings in the testCases.

Reply:
I had actually tried to use `keccak256(/*role*/)` as the `key` for the `Record`, but it wouldn't let me. I tried a couple of ways to do this, but to no avail. That is why I ended up using the actual value of `keccak256(/*role*/)` instead.

Reply:
I see the issue. The current version is fine then, resolving the conversation 👍

Reply:
I think using `[keccak256(/*role*/)]` as key will solve the problem, I mean, using the `[]`